zerver/views/scheduled_messages.py

```diff
@@ -36,7 +37,76 @@ def delete_scheduled_messages(
 
 
 @has_request_variables
-def scheduled_messages_backend(
+def update_scheduled_message_backend(
+    request: HttpRequest,
+    user_profile: UserProfile,
+    scheduled_message_id: int = REQ(converter=to_non_negative_int, path_only=True),
+    req_type: Optional[str] = REQ(
+        "type", str_validator=check_string_in(Message.API_RECIPIENT_TYPES), default=None
+    ),
+    req_to: Optional[str] = REQ("to", default=None),
+    topic_name: Optional[str] = REQ_topic(),
+    message_content: Optional[str] = REQ("content", default=None),
+    scheduled_delivery_timestamp: Optional[int] = REQ(json_validator=check_int, default=None),
+) -> HttpResponse:
+    if (
+        req_type is None
+        and req_to is None
+        and topic_name is None
+        and message_content is None
+        and scheduled_delivery_timestamp is None
+    ):
+        raise JsonableError(_("Nothing to change"))
```

Comment:
Added this error for situations where no scheduled message fields are updated. This is similar to a validation check we do in the message edit path.